cmake/helpers/GdalGenerateConfig.cmake

```diff
@@ -44,6 +44,11 @@ function(gdal_flatten_link_libraries _result)
         elseif(_lib STREQUAL "general")
             continue()
         endif()
+        if(_lib IN_LIST _seen_libs)
+            # Library has already been processed
+            continue()
+        endif()
+        list(APPEND _seen_libs "${_lib}")
```

Comment:
Here we are in exporting linker options (for gdal.pc).

The PR cuts away later references to a lib which has already been seen. This must not be done! You don't want to preserve the first `-lm` but the last.